DiceKit/MaximizationExpression.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  MaximizationExpression.swift
+//  DiceKit
+//
+//  Created by Logan Johnson on 8/15/15.
+//  Copyright © 2015 Brentley Jones. All rights reserved.
+//
+
+import Foundation
+
+public struct MaximizationExpression<Expression: protocol<ExpressionType, Equatable>>: Equatable {
+    
+    public let expression: Expression
+    
+    public init(_ expression: Expression) {
+        self.expression = expression
+    }
+    
+}
+
+// MARK: - ExpressionType
+
+extension MaximizationExpression: ExpressionType {
+    
+    public typealias Result = MaximizationExpressionResult<Expression.Result>
+    
+    public func evaluate() -> Result {
+        if let maximumOutcome = expression.probabilityMass.maximumOutcome() {
+            return Result(maximumOutcome)
+        }
+        
+        return Result(0)
+    }
+    
+    public var probabilityMass: ExpressionProbabilityMass {
+        if let maximumOutcome = expression.probabilityMass.maximumOutcome() {
+            return ProbabilityMass(maximumOutcome)
+        }
+        
+        return ProbabilityMass(FrequencyDistribution.additiveIdentity)
+    }
+    
+}
+
+// MARK: CustomStringConvertible
+
+extension MaximizationExpression: CustomStringConvertible {
+    
+    public var description: String {
+        return "max(\(expression))"
+    }
+    
+}
+
+// MARK: - CustomDebugStringConvertible
+
+extension MaximizationExpression: CustomDebugStringConvertible {
+    
+    public var debugDescription: String {
+        return "max(\(expression))"
+    }
+    
+}
+
+// MARK: - Equatable
+
+private func equate<E>(lhs: MaximizationExpression<E>, _ rhs: MaximizationExpression<E>) -> Bool {
+    return lhs.expression == rhs.expression && lhs.expression == rhs.expression
+}
+
+// When the same types check for commutative
+public func == <E>(lhs: MaximizationExpression<E>, rhs: MaximizationExpression<E>) -> Bool {
+    if equate(lhs, rhs) {
+        return true
+    }
+    
+    // Commutative
+    return lhs.expression == rhs.expression && lhs.expression == rhs.expression
```

Comment:
And actually, just do the one above compare here. Remove the `equate` method all together.